File: ch10/VB/App_Code/Listing10-9.vb
Imports Microsoft.VisualBasic

Public Class LastNameComparer
    Implements IComparer(Of String)

    Function Compare(x As String, y As String) As Integer _
        Implements IComparer(Of String).Compare

        Dim director1LastName As String
        Dim director2LastName As String
        director1LastName = x.Substring(x.LastIndexOf(" "))
        director2LastName = y.Substring(y.LastIndexOf(" "))
        Return director1LastName.CompareTo(director2LastName)
    End Function

End Class

File: ch10/CS/App_Code/Listing10-9.cs
using System.Collections.Generic;

public class LastNameComparer : IComparer<string>
{
    public int Compare(string x, string y)
    {
        var director1LastName = x.Substring(x.LastIndexOf(' '));
        var director2LastName = y.Substring(y.LastIndexOf(' '));
        return director1LastName.CompareTo(director2LastName);
    }
}